httpserver/server_certificates.h

```diff
@@ -86,9 +87,18 @@ int mkcert(X509** x509p, EVP_PKEY** pkeyp, int bits, int serial, int days) {
     } else
         x = *x509p;
 
-    rsa = RSA_generate_key(bits, RSA_F4, NULL, NULL);
+    bne = BN_new();
+    rsa = RSA_new();
```

Comment:
Similarly, does this need `RSA_free`?

Reply:
Needed to do some digging, but basically no. The rsa key is "attached" to the EVP_PKEY with `EVP_PKEY_assign_RSA`, so the EVP_PKEY takes care of freeing the rsa key...
https://www.openssl.org/docs/man1.0.2/man3/EVP_PKEY_assign_RSA.html